include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -57,34 +81,52 @@ class SplineBuilder
     using tag_type = typename InterpolationMesh::continuous_dimension_type;
 
 public:
+    /**
+     * @brief The type of the Kokkos execution space used by this class.
+     */
     using exec_space = ExecSpace;
 
+    /**
+     * @brief The type of the Kokkos memory space used by this class.
+     */
     using memory_space = MemorySpace;
 
     /**
-     * @brief The type of the interpolation mesh used by this class.
+     * @brief The type of the interpolation mesh (discrete dimension of interest) used by this class.
      */
     using interpolation_mesh_type = InterpolationMesh;
 
     /**
-     * @brief The type of the BSplines which are compatible with this class.
+     * @brief The BSplines dimension.
      */
     using bsplines_type = BSplines;
 
+	/**
+     * @brief The deriv dimension at the boundaries.
+     */
     using deriv_type = ddc::Deriv<tag_type>;
 
     /**
      * @brief The type of the domain for the interpolation mesh used by this class.
      */
     using interpolation_domain_type = ddc::DiscreteDomain<interpolation_mesh_type>;
 
+    /**
+     * @brief The type of the whole domain representing interpolation points.
+     */
     using batched_interpolation_domain_type = ddc::DiscreteDomain<IDimX...>;
```

Comment:
I'm not sure that the distinction between:

> The type of the whole domain representing interpolation points.

vs

> The type of the interpolation mesh (discrete dimension of interest) used by this class.

is clear for a new user. Maybe you could add an extended description to help clarify this

Reply:
Yes I think "mesh" is a bad name for a discrete dimension anyway because a mesh is bounded in its common sense, not a discrete dimension

Reply:
I agree that "mesh" is a bad name for a `ddc::DiscreteDomain<IDimX...>;` but maybe it makes sense in this context where there is only 1 valid mesh which is indeed bounded. The description doesn't need to describe the type generally just how it is used here. Probably it is preferred that the description is less general as the aim here is to clarify how the class works and how it should be used not how you can misuse it by also using `batched_interpolation_domain_type` for things for which it is not designed (but where `ddc::DiscreteDomain<IDimX...>;` would be valid).

Reply:
No, I mean mesh is a bad name for InterpolationMesh here (which is ie. ddc::UniformPointSampling<X>) because a UniformPointSampling is a mesh which goes from -infinity to +infinity, which is not the common concept of mesh